### PR TITLE
feat(hierarchicalMenu): add css class for link of selected menu item

### PR DESCRIPTION
**Pull request by dhayab**
**Summary**

This PR adds a new CSS class modifier and a key in `cssClasses` for the `hierarchicalMenu()` widget, following the addition made in the CSS specs.

| InstantSearch.css class name | Key in `cssClasses` |
| --- | --- |
| `.ais-HierarchicalMenu-link--selected` | `selectedItemLink` |

> **Note**
> In its current state, the widget does not conform to the InstantSearch.css specs. Since fixing this now introduces breaking changes to the DOM, I added a mention in the relevant component to address this when working on next major version.

[FX-1793](https://algolia.atlassian.net/browse/FX-1793)

### CODE REVIEW

File: src/widgets/hierarchical-menu/defaultTemplates.tsx [hunk 1]
```diff
@@ -6,9 +6,13 @@ import { cx } from '../../lib/utils';
...
 const defaultTemplates: HierarchicalMenuComponentTemplates = {
-  item({ url, label, count, cssClasses }) {
+  item({ url, label, count, cssClasses, isRefined }) {
+    const linkCssClasses = ([] as Array<string | undefined>)
+      .concat(cssClasses.link, isRefined ? cssClasses.selectedItemLink : [])
+      .filter(Boolean) as string[];
+
     return (
-      <a className={cx(cssClasses.link)} href={url}>
+      <a className={cx(linkCssClasses)} href={url}>
```
Comment (Haroenv) [inline src/widgets/hierarchical-menu/defaultTemplates.tsx:15]:
```suggestion
      <a className={cx([cssClasses.link, isRefined && cssClasses.selectedItemLink])} href={url}>
```

File: src/widgets/hierarchical-menu/defaultTemplates.tsx [hunk 2]
```diff
@@ -6,9 +6,13 @@ import { cx } from '../../lib/utils';
 import type { HierarchicalMenuComponentTemplates } from './hierarchical-menu';
 
 const defaultTemplates: HierarchicalMenuComponentTemplates = {
-  item({ url, label, count, cssClasses }) {
+  item({ url, label, count, cssClasses, isRefined }) {
+    const linkCssClasses = ([] as Array<string | undefined>)
+      .concat(cssClasses.link, isRefined ? cssClasses.selectedItemLink : [])
+      .filter(Boolean) as string[];
+
```
Comment (Haroenv) [inline src/widgets/hierarchical-menu/defaultTemplates.tsx:13]:
```suggestion
```

File: src/widgets/hierarchical-menu/defaultTemplates.tsx (hunk 1 above)
Comment (Haroenv) [reply src/widgets/hierarchical-menu/defaultTemplates.tsx:15]: see https://github.com/algolia/instantsearch.js/blob/68f3df412468fc4536201deb6eaa8bca0554911b/src/lib/utils/cx.ts#L3

**Review by Haroenv — CHANGES_REQUESTED** [review]
can be simplified a bit :)

**Review by sarahdayan — COMMENTED** [review]
Let's make sure to also:

- Add this new class to the other flavors
- Document the new class in all flavors
- Add this new class to the specs

File: src/widgets/hierarchical-menu/defaultTemplates.tsx (hunk 1 above)
Comment (dhayab) [reply src/widgets/hierarchical-menu/defaultTemplates.tsx:15]: `link` and `selectedItemLink` are typed as `string | string[] | undefined` so I need to concat them (or flatten the resulting array) to ensure we don't have arrays in arrays.

Comment (Haroenv) [reply src/widgets/hierarchical-menu/defaultTemplates.tsx:15]: could we call cx multiple times?

Comment (dhayab) [reply src/widgets/hierarchical-menu/defaultTemplates.tsx:15]: Oh that's a good idea! 👏

**Review by Haroenv — APPROVED** [review]

**Review by FabienMotte — APPROVED** [review]

**Review by sarahdayan — APPROVED** [review]

**Pull request merged**